formhero/forms/tests.py

```diff
@@ -1,4 +1,24 @@
-from django.test import TestCase
+from django.contrib.auth import get_user_model
+from formhero.apps.models import App
+from rest_framework.reverse import reverse
+from rest_framework.test import APITestCase
+from formhero.forms.views import CreateForm
+from rest_framework import status
 
 
-# Create your tests here.
+User = get_user_model()
+
+
+class RetrievePostFormTests(APITestCase):
+
+    def test_post_a_new_form(self):
+        app1 = App.objects.create()
+        handler = 'db'
+        data = {"app": app1.pk, "name": "Szymon", "handler": handler, "config": "test"}
+        url = reverse(CreateForm.URL_NAME, kwargs={'pk': 1})
+        response = self.client.post(url, data)
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(response.data['app'], app1.pk)
+        self.assertEqual(response.data['name'], "Szymon")
+        self.assertEqual(response.data['handler'], handler)
+        self.assertEqual(response.data['config'], "test")
```

Comment:
@szymonzmyslony Not sure why `handler` needs it's own variable declaration, but the strings for `app`, `name` and `config` are duplicated.

Also, we prefer single quotes to double wherever possible (pretty arbitrary rule, but it's just to encourage consistency).

Please stick to the convention of `test_should_...` for test method names. And the action you describe should refer to the system under test (SUT) which here is the **view**. So "create" is a better verb than "post", because the view is handling the POST rather than triggering it.

Here's how I'd write it...

```
def test_should_create_a_new_form(self):
    app = App.objects.create()
    post_data = {
        'app': app.pk,
        'name': 'Szymon',
        'handler: 'db',
        'config': 'test'
    }
    url = reverse(CreateForm.URL_NAME, kwargs={'pk': 1})

    response = self.client.post(url, post_data)
    
    self.assertEqual(response.status_code, status.HTTP_201_CREATED)
    self.assertEqual(response.data['app'], post_data['app'])
    self.assertEqual(response.data['name'], post_data['name'])
    self.assertEqual(response.data['handler'], post_data['handler'])
    self.assertEqual(response.data['config'], post_data['config'])
```

We might be able to get around writing an assert per key btw.

```
    # Alternatively, if the response is meant to exactly match, just do this...
    self.assertDictEqual(response.data, post_data)

    # Or, if post_data should be a subset of response.data...
    for key, value in post_data.items():
        self.assertEqual(response.data[key], value)
```

But don't worry too much about this, often it's better to manually list many assertions to be more explicit anyway.

Notice also that I've added line breaks between...

* setting up data
* making the request
* verifying the response

This is called Arrange, Act, Assert formatting. It's pretty useful to make the typical blocks in the test method clearly separated at a glance.

Mostly very picky comments here, and the actual test logic looks totally sound, so nice work! 👍